plan/physical_plans.go

```diff
@@ -105,10 +105,14 @@ func (p *physicalTableSource) addLimit(l *Limit) {
 }
 
 func (p *physicalTableSource) addTopN(prop *requiredProperty) bool {
+	if len(prop.props) == 0 && prop.limit != nil {
+		p.addLimit(prop.limit)
```

Comment:
If only add limit, the function name should not name as `addTopN`, or add limit should be called out side of `addTopN`.